plugin.video.imdb.trailers/addon.py

```diff
@@ -0,0 +1,326 @@
+# -*- coding: utf-8 -*-
+"""
+    IMDB Trailers Kodi Addon
+    Copyright (C) 2018 gujal
+    
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+# Imports
+from future import standard_library
+standard_library.install_aliases()
+from builtins import str
+from builtins import object
+import re
+import sys
+import urllib.parse
+import datetime
+import json
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import xbmcvfs
+import html.parser
+from bs4 import BeautifulSoup, SoupStrainer
+import requests
+import requests_cache
+
+# DEBUG
+DEBUG = False
+
+_addon = xbmcaddon.Addon()
+_addonID = _addon.getAddonInfo('id')
+_plugin = _addon.getAddonInfo('name')
+_version = _addon.getAddonInfo('version')
+_icon = _addon.getAddonInfo('icon')
+_fanart = _addon.getAddonInfo('fanart')
+_language = _addon.getLocalizedString
+_settings = _addon.getSetting
+PY3 = sys.version_info[0] == 3
+force_mode = _settings("forceViewMode") == "true"
+if force_mode:
+    menu_mode = int(_settings('MenuMode'))
+    view_mode = int(_settings('VideoMode'))
+
+CACHE_FILE = xbmc.translatePath('special://profile/addon_data/' + _addonID + '/requests_cache')
+requests_cache.install_cache(CACHE_FILE, backend='sqlite', expire_after=int(_settings('timeout'))*3600)
```

Comment:
The directory containing the file has to create first, otherwise an exception will be thrown.

Reply:
The directory containing the file should exist even on first start as it is the userdata addon folder containing the settings.xml
However as per comment, it has been moved below the force settings creation a few lines below so that the directory will certainly exist.

Reply:
No the directory is not automatically generated. Also not if the settings dialog is opened before.

Reply:
Yes, the directory doesn't get created if the user deliberately presses cancel on the settings dialog.
Thanks for that, Good Spotting!
Corrected now